app/main.py

```diff
@@ -1,25 +1,68 @@
-from abc import ABC
+from abc import ABC, abstractmethod
+from typing import Type
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.param_name = "_" + name
+
+    def __get__(self, instance: object, owner: Type) -> int:
+        return getattr(instance, self.param_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(f"Quantity should not be less than "
+                             f"{self.min_amount} and greater "
+                             f"than {self.max_amount}.")
+        setattr(instance, self.param_name, value)
 
 
 class Visitor:
-   pass
+    age = IntegerRange(min_amount=0, max_amount=120)
+    height = IntegerRange(min_amount=50, max_amount=250)
+    weight = IntegerRange(min_amount=10, max_amount=300)
+
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, visitor: Visitor) -> None:
+        self.visitor = visitor
+
+    @abstractmethod
+    def validate(self) -> bool:
+        pass
```

Comment:
As a result , that your validation logic should be in `__set__` method , this `validate` mathod can be deleted